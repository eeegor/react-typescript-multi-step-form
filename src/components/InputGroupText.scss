@import './InputGroup';

.input-text,
.input-email {
	display: block;
	width: 100%;
	padding: 0.7rem 0.75rem 0.5rem;
	font-size: 1rem;
	line-height: 2rem;
	color: #333;
	background-color: #fff;
	background-clip: padding-box;
	border: 1px solid #ccc;
	border-radius: 0.25rem;
	transition: border-color 0.15s ease-in-out, box-shadow 0.15s ease-in-out;
}
