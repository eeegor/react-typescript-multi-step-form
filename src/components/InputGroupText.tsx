/**
 * Component InputGroupText
 */

import * as React from 'react';
import * as classnames from 'classnames';
import './InputGroupText.scss';

export interface Props {
	id: string;
	name: string;
	value: string;
	label?: string;
	type?: 'text' | 'email';
	info?: string;
	status?: 'success' | 'danger';
	onChange?: (name: string, event: React.ChangeEvent<HTMLInputElement>) => void;
}

export interface State {}

export class InputGroupText extends React.Component<Props, State> {
	public static defaultProps: Partial<Props> = {
		type: 'text',
	};

	constructor(props: Props) {
		super(props);
	}

	render(): JSX.Element {
		const { id, value, onChange, label, type, name, status, info } = this.props;
		const cssClass = classnames(
			`input-group input-group--${type}`,
			status && `input-group--${status}`
		);

		return (
			<div id={id} className={cssClass}>
				{label && (
					<label className="input-group__label" htmlFor={name}>
						{label}
					</label>
				)}
				{info && <small className="input-group__info">{info}</small>}
				<input
					className="input-group__input input-text"
					id={name}
					name={name}
					type={type}
					onChange={event => onChange && onChange(name, event)}
					value={value}
				/>
			</div>
		);
	}
}
