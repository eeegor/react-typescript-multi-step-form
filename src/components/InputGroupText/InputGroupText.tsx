/**
 * Component InputGroupText
 */

import * as React from 'react';
import * as classnames from 'classnames';
import { InputGroup, Label, Info } from '..';
import './InputGroupText.scss';

export interface InputGroupTextProps {
	id: string;
	name: string;
	value?: string;
	label?: string;
	info?: string;
	type?: 'text' | 'email';
	status?: 'success' | 'danger';
	className?: string;
	required?: boolean;
	onChange?: (name: string, event: React.ChangeEvent<HTMLInputElement>) => void;
}

export class InputGroupText extends React.Component<InputGroupTextProps> {
	render(): JSX.Element {
		const {
			id,
			className,
			required,
			value,
			onChange,
			label,
			type,
			name,
			status,
			info,
		} = this.props;

		return (
			<InputGroup id={id} type={type} status={status} className={classnames(className)}>
				<Label className="input-group__label" htmlFor={id || name} label={label} />
				<Info className="input-group__info" label={info} />
				<input
					className={`input-group__input input-${type || 'text'}`}
					id={name}
					name={name}
					type={type}
					onChange={event => onChange && onChange(name, event)}
					value={value || ''}
					autoFocus={true}
					required={required}
				/>
			</InputGroup>
		);
	}
}
