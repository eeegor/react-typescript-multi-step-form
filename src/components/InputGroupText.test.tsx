import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { InputGroupText } from './InputGroupText';
import { mount } from 'enzyme';

describe('InputGroupText', () => {
	const callback = jest.fn();
	const testData = {
		id: 'name',
		name: 'name',
		label: 'Your Name',
		value: 'Bob',
		onChange: callback,
	};

	it('renders without crashing', () => {
		const div = document.createElement('div');
		ReactDOM.render(<InputGroupText />, div);
		ReactDOM.unmountComponentAtNode(div);
	});

	it('change event works', () => {
		const wrapper = mount(<InputGroupText {...testData} />);
		wrapper.find('.input-group__input').simulate('change');
		expect(callback).toHaveBeenCalledTimes(1);
	});

	it('should adopt a valid info attribute', () => {
		const info = 'test';
		const wrapper = mount(<InputGroupText info={info} {...testData} />);
		expect(wrapper.find('.input-group__info').text()).toEqual('test');
	});

	it('should adopt a valid status attribute', () => {
		const status = 'danger';
		const wrapper = mount(<InputGroupText status={status} {...testData} />);
		expect(wrapper.find('.input-group').hasClass('input-group--danger')).toEqual(true);
	});
});
