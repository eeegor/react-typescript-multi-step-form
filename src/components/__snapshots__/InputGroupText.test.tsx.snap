// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputGroupText renders markup correctly 1`] = `
<div
  className="input-group input-group-text"
  id="name"
>
  <label
    className="input-group__label"
    htmlFor="name"
  >
    Your Name
  </label>
  <input
    className="input-group__input input-text"
    id="name"
    name="name"
    onChange={[Function]}
    type="text"
    value="Bob"
  />
</div>
`;
